src/preprocess.py

```diff
@@ -0,0 +1,154 @@
+import hyperparameters as hp
+import pandas as pd
+from torch.utils.data import Dataset, DataLoader
+import os
+import librosa
+import numpy as np
+from data import raw_text_to_phoneme_ids
+import collections
+from scipy import signal
+import torch as t
+import math
+
+
+class LJDatasets(Dataset):
+    """LJSpeech dataset."""
+
+    def __init__(self, csv_file, root_dir):
+        """
+        Args:
+            csv_file (string): Path to the csv file with annotations.
+            root_dir (string): Directory with all the wavs.
+
+        """
+        self.landmarks_frame = pd.read_csv(csv_file, sep='|', header=None)
+        self.root_dir = root_dir
+
+    def load_wav(self, filename):
+        return librosa.load(filename, sr=hp.sample_rate)
+
+    def __len__(self):
+        return len(self.landmarks_frame)
+
+    def __getitem__(self, idx):
+        wav_name = os.path.join(self.root_dir, self.landmarks_frame.ix[idx, 0]) + '.wav'
+        original_text = self.landmarks_frame.ix[idx, 1]
+
+        text = np.asarray(raw_text_to_phoneme_ids(original_text), dtype=np.int32)
+        mel = np.load(wav_name[:-4] + '.pt.npy')
+        mel_input = np.concatenate([np.zeros([1,hp.num_mels], np.float32), mel[:-1,:]], axis=0)
+        text_length = len(text)
+        pos_text = np.arange(1, text_length + 1)
```

Comment:
What exactly are the "pos_*" variables used for?  Is this for our network output?

Reply:
They are indices basically. They are left over from the other repo. Could be useful for training? But I think they can mostly be ignored. Thought it'd be nice to leave in case we use it in the future.